### DISCUSSION TITLE
next build is ignoring redirects in next.config.js and using wrong locale

### DISCUSSION THREAD

**Discussion by kristijorgji** [vercel/next.js#29003] [Help]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

v15.0.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

yarn start (next dev)

### Describe the Bug

The redirects specified in next.config.js, are ignored by next build and the js files are not created for the redirected routes.
I have these configs:

next-i18next.config.js
```
const path = require('path');

module.exports = {
    localePath: path.resolve('./public/static/locales'),
    i18n: {
        defaultLocale: 'en_US',
        locales: ['en_US', 'de_DE'],
    },
};

```

next.config.js
```
const { i18n } = require('./next-i18next.config');

/** @type {import('next/dist/server/config').NextConfig} */
module.exports = {
    i18n,
    async rewrites() {
        return [
            {
                source: '/de/anmelden',
                destination: '/de_DE/login',
            }
        ];
    },
};

```

folder structure:
pages/login.tsx

Accessing the site like localhost:3000/de_DE/login works great, also without de_DE /login gives site back in english.

Now I want to have specific name for de_DE of this url, like /de/anmelden and I used redirects as there is no native support for this.

Accessing localhost:3000/de/anmelden works fine initially, i can see the site in german. Then if i want to change language or interact with the page, i get "/pages/de/anmelden.js" not found. Also if i want ot change locale, i get with react router this url `http://localhost:3000/de_DE/de/anmelden`

it seems like next thinks that the locale is in english somehow, and makes a wrong redirect

### Expected Behavior

in .next, the content for /de/anmelden.json /de/anmelden.html are build instead of de_DE/login.html|json

Also should work fine with react router
Is there any way for router to form href full urls from pathname ? I could not find such.
Using the next.config.js redirects works okish in server side, but in the client the router redirects to the not redirected url, also in no way to generate the url with react router

### To Reproduce

Provided in bug description config of next-i18n and next.config.js

**Comment by smo043** [comment]
I dont have locales, but having redirect rewrites which is failing now. Please help.

**Reply by smo043** [reply]
@ijjk - This is my configuration without locale and it is also failing. First it redirects to my-info and immediately it replaces the my-info with undefined.



```
async rewrites() {
    return [
      {
        source: '/apt/phn',
        destination: '/api/phn'
      },
      {
        source: '/hp',
        destination: '/api/hp'
      }
    ];
  },
  async redirects() {
    return [
      {
        source: '/app',
        destination: '/app/my-info',
        permanent: false
      }
    ];
  }
  ```

**Reply by ijjk** [reply]
Can you provide a repo with a minimal reproduction and the specific URLs that aren't working as expected?

**Comment by ijjk** [comment]
Hi, the above redirect does not work as you are specifying the locale in the `source` and `destination` without setting `locale: false` by default the locales are matched for you and will pass it through. See more information here https://nextjs.org/docs/api-reference/next.config.js/redirects#redirects-with-i18n-support

**Reply by smo043** [reply]
@ijjk - Issue was with improper env variables, it is now working as expected. This can be closed.

**Reply by kristijorgji** [reply]
> @ijjk - Issue was with improper env variables, it is now working as expected. This can be closed.

Wait, please explain the solution
